### DISCUSSION TITLE
Power factor interpretation

### DISCUSSION THREAD

**Discussion by CSH12** [WillCodeForCats/solaredge-modbus-multi#462] [Q&A]
Hi @all, 

Does anyone knows how to interprete the power factor data? 
I have had a look at it since yesterday and saw this: 
![image](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/97809992/43bcfd29-2145-4dbe-8083-8e5e839c3a76)

I'm not really sure what that means to me? 
Normally I would assume that the power factor is a number between 0...1 or 0...100% but why it is possible to have it in both directions? 
And also in the period of receiving energy from grid (1:30 to 7:30 am), why it's having round about 50%?

My originally question for thinking about the power factor was: why the battery power (DC) is higher than the inverter power (house consumtion in AC). At best looking to the period from e.g. from 0:00 to 1:30 am (distance to 0 is not equidistant)
![image](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/97809992/62cf2e28-25c9-4f72-aeb2-fcdeb4126384)

--> maybe there is a context with power factor and have I got to multiply the power factor to the battery's DC power so that it matches? but actually I don't see the context if there is one...

If anyone has some answers, I would appreciate to know. 

Thanks
Chris

**Comment by WillCodeForCats** [comment]
Power factor is simply the ratio of W (real power) to VA (apparent power). Positive is leading, negative is lagging, and the middle point is "unity" power factor.

Usually you see power factor represented as 1-0 and with 0.80 as a nominal value for rating things like VA to W on a UPS. The inverters return +- 100-0% so you can just divide it by 100 to see the "traditional" number (i.e. 80% is 0.80, 100% is unity). Power factor doesn't have units since it's a ratio. The graph unfortunately looks weird because in this application it PF should be centered around 100 (100/100 = 1 unity) in the middle and not 0.

At very large industrial scale utilities often require the consumer to install power factor correction. At residential levels it's irrelevant other than a fun statistic. Poor power factors are energy wasted (var).

Active power (W, watts) is the "good" part that's doing useful work. Reactive power (var, volt-amps reactive) is lost power. Apparent power (VA, volt-amps) is the total demand power. Since power factor is your ratio of working to lost power, a PF of 0.5 (or 50%) means it's doing something very inefficient during that time. If you were to turn on the VA and var sensors you would probably see that VA is double the W.

* https://www.fluke.com/en-us/learn/blog/power-quality/power-factor-formula
* https://en.wikipedia.org/wiki/Power_factor#Lagging,_leading_and_unity_power_factors

**Comment by CSH12** [comment]
Hi @WillCodeForCats, 

Thanks for the quick reply and the explanation 
Think I got it (mostly ;-) )

![2023-11-01_20h46_50](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/97809992/df808618-7c86-42b2-8375-604ca5ff6219)
In my example from 0:00 to 1:30 it would mean: 
- PF = -18.xx% (1 - 18.xx%/100%) = 0.81 
- B1_DC_Power * PF ~= I1_AC_Power --> more or less **B1_AC_Power** as real power

I'm asking because the "House Consumtion" is shown as the B1_DC_Power in the energy flow - but that is not the correct "House Consumption power
![image](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/97809992/8d4ae56d-c88d-4276-86ef-e875f740e48c)
If I click on the House Consumtion I just see a power of ~4xx W in that case because of the PF lag. Otherwise it is not the same working point and I compare apples with pears.

(this might be also a good hint for your config template example. The data are not the same base. For the powerflow it has to be adapted) 

Thanks 
Chris

**Reply by WillCodeForCats** [reply]
I don't actually use any templates with my setup or have batteries, so the templates were contributed but I haven't personally looked at or tested them.

Power factor of 18% is 0.18 in the 1-0 scale.

**Reply by CSH12** [reply]
Ok. was just a hint. maybe it will help others... I can see if I'm able to comment that. Let's see.

but that 18% resp. 0.18 is not plausible to me. 
then battery DC power has to be much bigger (~2260W). this will result in ~411W Inverter AC Power with a PF of 0.18.
And also an efficiency of 18% would be very poor for a new battery ;-)
82% would be more realistic... don't you also think so?

**Reply by WillCodeForCats** [reply]
Start tracking your AC VA sensor and compare it to AC power (watts).

Sensors `sensor.solaredge_i1_ac_va` and `sensor.solaredge_i1_ac_power`
